core/interval/int.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package interval
+
+import (
+	"sort"
+)
+
+type IntegerInterval struct {
+	Lower int
```

Comment:
These should be uints, right now they could be negative.

Reply:
Since the intervals are in core, they should be generic. it's possible another application could require negative numbers in their intervals

Really this should be enforced on PortRange